### ISSUE TITLE
TypeError: str.replace is not a function. 

### ISSUE THREAD

**Issue by TimiBolu** [icebob/fastest-validator#344]
Hi, I really appreciate your library and have used it ever since I found out about it (roughly a year). 

However, I encountered an issue. 
When I tried to compile the following code, I get the error below,

*** The code ****

<img width="1116" alt="Screenshot 2024-05-24 at 10 51 06" src="https://github.com/icebob/fastest-validator/assets/33407288/8d67cdc8-f019-4302-8cd8-fbbc9560ff7c">

**** The error ****

<img width="1116" alt="Screenshot 2024-05-24 at 10 45 56" src="https://github.com/icebob/fastest-validator/assets/33407288/08962522-11ce-47aa-b33b-c9ccc3ad7ea8">

When I change any two things, it works fine.
1 - If I remove the "form" field. 
or 
2 - or I change the string types
<img width="1116" alt="Screenshot 2024-05-24 at 10 47 47" src="https://github.com/icebob/fastest-validator/assets/33407288/e62a8dfb-c85b-4f08-9508-66586be491ec">

Please what could be the issue.

**Comment by TimiBolu** [comment]
```
import Validator from "fastest-validator";

const v = new Validator();

const header = {
    "socials": {
        "$$type": "object|optional",
        "apps": {
            "type": "array",
            "items": {
                "type": "object",
                "props": {
                    "name": {
                        "type": "string",
                        "enum": [
                            "instagram",
                            "twitter",
                            "facebook",
                            "github",
                            "linkedin",
                            "tiktok",
                            "dribbble",
                            "behance",
                            "twitch",
                            "snapchat",
                            "email",
                            "whatsapp",
                            "webaddress"
                        ]
                    },
                    "handle": {
                        "type": "string",
                        "optional": true
                    },
                    "url": {
                        "type": "string",
                        "optional": true
                    }
                }
            },
            "optional": true
        },
        "iconStyle": {
            "type": "string",
            "optional": true,
            "enum": [
                "outline",
                "color",
                "solid"
            ]
        },
        "alignment": {
            "type": "string",
            "optional": true,
            "enum": [
                "left",
                "center",
                "right"
            ]
        }
    },
    "text": {
        "$$type": "object|optional",
        "value": {
            "type": "string",
            "optional": true
        },
        "font": {
            "type": "string",
            "optional": true
        }
    },
    "title": {
        "$$type": "object|optional",
        "heading": {
            "type": "string",
            "optional": true
        },
        "subHeading": {
            "type": "string",
            "optional": true
        },
        "alignment": {
            "type": "string",
            "optional": true,
            "enum": [
                "left",
                "center",
                "right"
            ],
            "default": "center"
        }
    },
    "image": {
        "$$type": "object",
        "value": {
            "type": "string",
            "optional": true
        },
        "style": {
            "type": "string",
            "optional": true,
            "enum": [
                "round",
                "straight",
                "curved",
                "curved-md"
            ]
        },
        "alignment": {
            "type": "string",
            "optional": true,
            "enum": [
                "full_width",
                "none"
            ]
        }
    },
    "layout": {
        "type": "string",
        "enum": [
            "img-right",
            "img-left",
            "img-top",
            "img-bg",
            "img-top-bg"
        ]
    },
    "form": {
        "$$type": "object|optional",
        "title": {
            "type": "string"
        },
        "description": {
            "type": "string"
        },
        "label": {
            "type": "string"
        },
        "field1": {
            "$$type": "object",
            "name": {
                "type": "string"
            },
            "value": {
                "type": "string"
            }
        },
        "field2": {
            "$$type": "object",
            "name": {
                "type": "string"
            },
            "value": {
                "type": "string"
            }
        }
    }
};

const ssc = v.compile(header);
console.log({ssc})
```

**Comment by TimiBolu** [comment]
```
const data =  {
    "image": {
        "value": "https://res.cloudinary.com/dnsbjog0b/image/upload/v1705415231/fmm2oskwp6r2q8gu1ack.jpg",
        "style": "round",
        "alignment": "none"
    },
    "text": {
        "size": "narrow",
        "value": "With fingers flying over the keyboard, I turn code into symphonies, creating not just functional but stylish tech wonders."
    },
    "title": {
        "heading": "Timi Adesina",
        "subHeading": "",
        "alignment": "center"
    },
    "socials": {
        "apps": {
        "twitch": {
            "handle": "",
            "url": "",
            "_id": {
            "$oid": "6672cf22587f0ca5deab6005"
            }
        },
        "instagram": {
            "handle": "",
            "url": "https://www.instagram.com/",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ffa"
            }
        },
        "twitter": {
            "handle": "TimmyAdesina",
            "url": "https://twitter.com/TimmyAdesina",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff9"
            }
        },
        "facebook": {
            "handle": "",
            "url": "https://web.facebook.com/",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff8"
            }
        },
        "github": {
            "handle": "TimiBolu",
            "url": "https://github.com/TimiBolu",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff7"
            }
        },
        "linkedin": {
            "handle": "timiadesina",
            "url": "https://www.linkedin.com/in/timiadesina",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff6"
            }
        },
        "tiktok": {
            "handle": "",
            "url": "https://www.tiktok.com/@",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff5"
            }
        },
        "dribbble": {
            "handle": "",
            "url": "https://dribbble.com/",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff4"
            }
        },
        "behance": {
            "handle": "",
            "url": "https://www.behance.net/",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff3"
            }
        },
        "snapchat": {
            "handle": "",
            "url": "https://www.snapchat.com/add/",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff2"
            }
        },
        "email": {
            "handle": "adesinatim@gmail.com",
            "url": "mailto:adesinatim@gmail.com",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff1"
            }
        },
        "whatsapp": {
            "handle": "09066744819",
            "url": "https://wa.me/09066744819",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5ff0"
            }
        },
        "webaddress": {
            "handle": "",
            "url": "//",
            "_id": {
            "$oid": "6672cf22587f0ca5deab5fef"
            }
        }
        },
        "iconStyle": "outline",
        "alignment": "center"
    },
    "form": {
        "title": "",
        "description": "",
        "field1": "",
        "field2": "",
        "label": ""
    },
    "layout": "img-top",
}
```

**Comment by FerX** [comment]
ok, the issue is "label" property into form..

try this:
```
"form": {
type:"object",
props:{
	"title": {
		"type": "string"
	},
	"description": {
		"type": "string"
	},
	"label": {
		"type": "string"
	}
}
}
```

**Comment by FerX** [comment]
the pull request https://github.com/icebob/fastest-validator/pull/345
fix also this bug.

**Comment by TimiBolu** [comment]
Thank you so much. I really appreciate it. 
I actually wrote a very light wrapper for my use case with a joi/zod like syntax that compiles to fastest-validator schema. 
I have discovered writing the json schema like syntax can be quite daunting so I now auto generate them. 
It kind of allows me get the best of both worlds.
I feel this may be a good way forward as an api extension for users who may be interested.
